Make Routing Setup Clearer

gary-kim opened the pull request:
Closes #110 
Screenshots
![Screenshot from 2019-10-22 16-53-57](https://user-images.githubusercontent.com/47195730/67271145-9291d200-f4ec-11e9-9949-756a66667198.png)

jancborchardt commented on templates/adminSettings.php:
```diff
@@ -9,10 +9,10 @@
-    <label><?php p($l->t('An OSRM server URL looks like this : https://my.osrm.server.org:5000/route/v1')); ?></label><br/>
-    <label><?php p($l->t('Leave URL fields empty to disable OSRM routing provider.')); ?></label><br/>
-    <br />
+	<h4><?php p($l->t('To enable routing, you must set up a routing engine below.')) ?></h4>
```
This should probably be a `p class="settings-hint"` instead of a "h4", as it’s not a heading. :)

jancborchardt commented on templates/adminSettings.php:
```diff
@@ -28,21 +28,19 @@
-    <label><?php p($l->t('Set the API key to use Mapbox routing service.')); ?></label><br/>
-    <label><?php p($l->t('Leave empty to disable.')); ?></label><br/>
-    <br/>
+	<h3><a href="https://www.mapbox.com/" target="_blank"><?php p($l->t('Mapbox settings')); ?></a></h3>
```
Same here with whitespace

jancborchardt commented on templates/adminSettings.php:
```diff
@@ -28,21 +28,19 @@
-    <br/>
+	<h3><a href="https://www.mapbox.com/" target="_blank"><?php p($l->t('Mapbox settings')); ?></a></h3>
+    <p><?php p($l->t('Set the API key to use Mapbox routing service.')); ?></p>
+    <p><?php p($l->t('Leave empty to disable.')); ?></p><br/>
```
These 2 paragraphs could be in one for simplicity?

jancborchardt replied:
Or is this intentional like with OSRM above to not make the sentence so long? Does make sense, so your call @gary-kim :)

gary-kim replied:
Left it as two paragraphs because it feels like two distinct messages and consistency.

jancborchardt commented:
@jakobroehrl @Aquariu @WNYmathGuy if you could test if this fixes https://github.com/nextcloud/maps/issues/110 for you, that would be great. :) If you don’t have a testing instance that’s ok, nevermind in that case.

gary-kim commented on templates/adminSettings.php:
```diff
@@ -28,21 +28,19 @@
-    <label><?php p($l->t('A GraphHopper server URL looks like this : https://my.graphhopper.server.org:8989/route')); ?></label>
-    <br />
-    <br />
+	<h3><a href="https://www.graphhopper.com/" target="_blank"><?php p($l->t('GraphHopper settings')); ?></a></h3>
```
This is what I get for not syncing IDE settings. Fixed.

jakobroehrl commented:
Can't test it, but I think it looks good

jancborchardt commented:
(Just btw @gary-kim since you are member of the Nextcloud organization, you can work in branches of the main repositories and don’t need to fork them and work in these. That makes collaboration easier as others can add pull requests too for small changes, and also you don’t have all the hassle of keeping your fork updated. :)

eneiluj commented:
Nice! Thanks a lot @gary-kim! Looks good!

While we're on this, the routing engine selector gets too wide when it content grows and it goes under the "invert route" and "add step" buttons. Could you do something about it?

By the way, those buttons would need a "title".

If you cancel this PR and push your branch to the nextcloud/maps repo, I'll be able to add those changes myself if you want.

gary-kim commented:
@eneiluj Addressed the issues you talked about. What do you think?
Screenshot:
![Screenshot from 2019-10-28 14-38-14](https://user-images.githubusercontent.com/47195730/67657056-9ad88a00-f990-11e9-9a90-8c8193ddc65a.png)

tacruc commented:
@gary-kim can you remove the fixup commit. (Guide is in the fixupbot check)

gary-kim commented:
I usually squash fixup commits when ready to merge. Should we merge now?

jancborchardt commented:
Very nice enhancement @gary-kim! :)
